pkg/ebpf/tracee.go

```diff
@@ -541,20 +543,21 @@ func (t *Tracee) Init(ctx gocontext.Context) error {
 	// Checking the kernel symbol needs to happen after obtaining the capability;
 	// otherwise, we get a warning.
 	usedClockID := traceetime.CLOCK_BOOTTIME
-	err = capabilities.GetInstance().Specific(
+	err = capabilities.GetInstance().EBPF(
 		func() error {
+			supported, err := bpf.BPFHelperIsSupported(bpf.BPFProgTypeKprobe, bpf.BPFFuncKtimeGetBootNs)
+
+			// only report if operation not permitted
+			if errors.Is(err, syscall.EPERM) {
+				return err
```

Comment:
To avoid shadowing and confusion. `e` is just a suggestion, check if it also doesn't shadow.

```suggestion
			supported, e := bpf.BPFHelperIsSupported(bpf.BPFProgTypeKprobe, bpf.BPFFuncKtimeGetBootNs)

			// only report if operation not permitted
			if errors.Is(e, syscall.EPERM) {
				return err
```

Reply:
good catch, thank you for the suggestion but think `innerErr` would be better, WDYT?